backend/src/BuiltinExecution/Libs/NoModule.fs

```diff
@@ -152,7 +152,8 @@ and equalsExpr (expr1 : Expr) (expr2 : Expr) : bool =
     equalsExpr target1 target2
     && NEList.length cases1 = NEList.length cases2
     && NEList.forall2
-      (fun (p1, e1) (p2, e2) -> equalsMatchPattern p1 p2 && equalsExpr e1 e2)
+      (fun case1 case2 ->
+        equalsMatchPattern case1.pat case2.pat && equalsExpr case1.rhs case2.rhs)
```

Comment:
probably need an additional check here for `equalsSomethingOptional case1.whenCondition case2.whenCondition`?